Sources/BraintreeVisaCheckout/BTVisaCheckoutClient.swift

```diff
@@ -1,6 +1,137 @@
+import VisaCheckoutSDK
 import Foundation
 
-public class BTVisaCheckoutClient {
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
 
-    // TODO: - Add implementation
+/// Creates a Visa Checkout client for processing Visa Checkout payments.
+@objc public class BTVisaCheckoutClient: NSObject {
+
+    private let apiClient: BTAPIClient
+
+    /// Creates a Visa Checkout client.
+    ///
+    /// - Parameters:
+    ///   - apiClient: An API client.
+    @objc(initWithAPIClient:)
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+    }
+
+    /// Creates a Visa Checkout `Profile` with the merchant API key, environment, and other properties to be used with Visa Checkout.
+    ///
+    /// - Parameters:
+    ///   - completion: A completion block that is invoked when the profile is created. If the profile creation succeeds,
+    ///     `Profile` will contain a profile and `error` will be `nil`.
+    ///     If it fails, `Profile` will be `nil` and `error` will describe the failure.
+    @objc public func createProfile(completion: @escaping (Profile?, Error?) -> Void) {
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            if let error {
+                /// TODO: Add failure analytics event
+                completion(nil, error)
+                return
+            }
+
+            guard let configuration else {
+                completion(nil, BTVisaCheckoutError.fetchConfigurationFailed)
+                return
+            }
+
+            guard configuration.isVisaCheckoutEnabled else {
+                completion(nil, BTVisaCheckoutError.disabled)
+                return
+            }
+
+            guard let visaCheckoutAPIKey = configuration.visaCheckoutAPIKey else {
+                completion(nil, BTVisaCheckoutError.integration)
+                return
+            }
+
+            let environmentString = configuration.visaCheckoutEnvironment
+            let environment: Environment = environmentString == "sandbox" ? .sandbox : .production
+
+            let profile = Profile(
+                environment: environment,
+                apiKey: visaCheckoutAPIKey,
+                profileName: nil
+            )
+            profile.datalevel = .full
+            profile.clientId = configuration.visaCheckoutExternalClientID
+            profile.acceptedCardBrands = configuration.acceptedCardBrands
+
+            completion(profile, nil)
+        }
+    }
+
+    /// Tokenizes a Visa checkout result.
+    /// - Note: The `checkoutResult` parameter is declared as `callID` type, but you must pass a `VisaCheckoutResult` instance.
```

Comment:
Looks like this should be updated to the following based on the method signature:
```suggestion
    /// - Note: The `checkoutResult` parameter is declared as `callID` type, but you must pass a `CheckoutResult` instance.
```